src/components/connect.tsx

```diff
@@ -736,7 +745,10 @@ function connect<
 
       try {
```

Comment:
q: when this can throw with `useSyncExternalStore`? shouldn't `useSyncExternalStore` be resilient to tearing etc that could make this sort of stuff throw pre-uSES?